scipy/stats/tests/test_distributions.py

```diff
@@ -913,6 +913,93 @@ def test_sf(self):
         expected = (scale/x)**b   # 2.25e-18
         assert_allclose(p, expected)
 
+    @pytest.mark.filterwarnings("ignore:invalid value encountered in "
+                                "double_scalars")
+    @pytest.mark.parametrize("rvs_shape", [1, 2])
+    @pytest.mark.parametrize("rvs_loc", [0, 2])
+    @pytest.mark.parametrize("rvs_scale", [1, 5])
+    def test_fit(self, rvs_shape, rvs_loc, rvs_scale):
+        data = stats.pareto.rvs(size=100, b=rvs_shape, scale=rvs_scale,
+                                loc=rvs_loc)
+        # non fixed `floc` does not use MLE.
+        all_free = stats.pareto.fit(data)
+        all_free_opt = super(type(stats.pareto), stats.pareto).fit(data)
+        assert_equal(all_free, all_free_opt)
```

Comment:
@WarrenWeckesser It looks like these just check that the right code - analytical or numerical fit - is being run depending on the input parameters, and it's difficult to tell directly from the outside. Thoughts about them? Is it OK to have these sort-of indirect tests that the intended code runs?

Reply:
It occurs to me that if in `test_fit_MLE_comp_optimzer` the `<` is strictly satisfied, then that is enough to confirm that `fit` is running the analytical MLE code. So I'd suggest including just the `<` and _not_ `np.allclose` wherever possible, and then eliminating the tests up here as much as possible. 

Actually - and I know this is probably controversial - if the difference in the execution time for the analytical vs numerical optimization is big enough, then you could actually measure the execution time for each of the tests in `test_fit_MLE_comp_optimzer` and use that to determine whether the analytical code ran. That would let you get rid of all of these.

Reply:
With the changes Warren suggested to never fall back on the optimizer, it removes the need to test whether or not analytical MLEs are being used. Given that it still passes, should I still remove `np.allclose`?